source/sdk/Buffer.ooc

```diff
@@ -0,0 +1,78 @@
+/*
+* Copyright (C) 2015 - Simon Mika <simon@mika.se>
+*
+* This sofware is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 2.1 of the License, or (at your option) any later version.
+*
+* This software is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this software. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+Buffer: cover {
+	_pointer: Pointer
+	_size: Int
+	pointer ::= this _pointer
+	size ::= this _size
+	init: func@ { this init(null, 0) }
+	init: func@ ~allocate (size: Int) { this init(malloc(size), size) }
+	init: func@ ~fromData (=_pointer, =_size)
+	free: func@ -> Bool {
+		result := this _pointer != null && this _size != 0
+		if (result) {
+			free(this _pointer)
+			this _pointer = null
+			this _size = 0
+		}
+		result
+	}
+	extend: func@ (size: Int) {
+		newPointer := this _pointer
+		if (size > this _size) {
```

Comment:
How about changing this method name to `resize` and this condition to `if (size != this _size)` ?